processor/k8sattributesprocessor/e2e_test.go

```diff
@@ -12,60 +12,207 @@
+				"container.image.tag":         newExpectedValue(equal, "latest"),
+				"container.id":                newExpectedValue(exist, ""),
+			},
+		},
+		{
+			name:    "daemonset",
+			service: "test-daemonset",
+			attrs: map[string]*expectedValue{
+				"k8s.pod.name":                newExpectedValue(regex, "test-telemetrygen-daemonset-[a-z0-9]*"),
+				"k8s.pod.ip":                  newExpectedValue(exist, ""),
+				"k8s.pod.uid":                 newExpectedValue(exist, ""),
+				"k8s.pod.start_time":          newExpectedValue(exist, ""),
+				"k8s.node.name":               newExpectedValue(exist, ""),
+				"k8s.namespace.name":          newExpectedValue(equal, "default"),
+				"k8s.daemonset.name":          newExpectedValue(equal, "test-telemetrygen-daemonset"),
+				"k8s.daemonset.uid":           newExpectedValue(exist, ""),
+				"k8s.annotations.workload":    newExpectedValue(equal, "daemonset"),
+				"k8s.labels.app":              newExpectedValue(equal, "test-telemetrygen-daemonset"),
+				"k8s.container.name":          newExpectedValue(equal, "telemetrygen"),
+				"k8s.container.restart_count": newExpectedValue(equal, "0"),
+				"container.image.name":        newExpectedValue(equal, "ghcr.io/open-telemetry/opentelemetry-collector-contrib/telemetrygen"),
+				"container.image.tag":         newExpectedValue(equal, "latest"),
+				"container.id":                newExpectedValue(exist, ""),
+			},
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			scanSpanForAttributes(t, tc.service, tc.attrs)
+		})
+	}
+}
+
+func scanSpanForAttributes(t *testing.T, expectedService string, kvs map[string]*expectedValue) {
+	var err error
+
+	f, _ := os.Open("./testdata/trace.json")
 	defer f.Close()
 
 	scanner := bufio.NewScanner(f)
-
 	const maxCapacity int = 8388608
 	buf := make([]byte, maxCapacity)
 	scanner.Buffer(buf, maxCapacity)
 
 	unmarshaler := ptrace.JSONUnmarshaler{}
-	jobKeyFound := false
-	nsKeyFound := false
 	for scanner.Scan() {
-		traces, err := unmarshaler.UnmarshalTraces(scanner.Bytes())
-		assert.NoError(t, err)
+		traces, serr := unmarshaler.UnmarshalTraces(scanner.Bytes())
+		if serr != nil {
+			continue
+		}
 
 		len := traces.ResourceSpans().Len()
 		for i := 0; i < len; i++ {
 			resourceSpans := traces.ResourceSpans().At(i)
 
-			resourceSpans.Resource().Attributes().Range(
-				func(k string, v pcommon.Value) bool {
-					if k == "k8s.job.name" && v.Str() == "test-telemetrygen" {
-						jobKeyFound = true
-					}
-					if k == "k8s.namespace.name" && v.Str() == "default" {
-						nsKeyFound = true
-					}
-					return true
-				},
-			)
+			service, exist := resourceSpans.Resource().Attributes().Get("service.name")
+			assert.Equal(t, true, exist, "span do not has 'service.name' attribute in resource")
+
+			if service.AsString() != expectedService {
+				continue
+			}
+
+			if err = resourceHasAttributes(resourceSpans.Resource(), kvs); err == nil {
+				return
```

Comment:
If we exit with nil, I think we should exit for any err as well, otherwise, we will be flooding the error output with duplicate messages